Develop

RadicalZephyr opened the pull request.

RadicalZephyr committed:
Add more log sweeping configurations to StackConfiguration

Add logging s3 bucket to stack configuration
Add flag to stack config to control deleting logs post-sweep

RadicalZephyr committed:
Simpler incarnation of the TimeBasedRollingToS3Policy

RadicalZephyr committed:
Add SweepAction class

RadicalZephyr committed:
Add StackConfigAccess class

RadicalZephyr committed:
Add AmazonS3Provider class

RadicalZephyr committed:
Small cleanups to test

RadicalZephyr committed:
Move to the rolling policy simply generating a UUID on activation

This simplifies things, and it's very clear that the UUID is not nearly
long enough to cause problems.

RadicalZephyr committed:
Test filling out

brucehoff commented on services/repository/src/test/java/org/sagebionetworks/sweeper/log4j/SweepActionTest.java:
```diff
@@ -0,0 +1,115 @@
+		SweepAction sweepAction = new SweepAction(mockFile, "", "", mockS3Client, false);
+
+		when(mockFile.exists()).thenReturn(true);
+		sweepAction.execute();
```
Check that mockS3Client archives mockFile

brucehoff commented on services/repository/src/main/java/org/sagebionetworks/sweeper/log4j/TimeBasedRollingToS3Policy.java:
```diff
@@ -0,0 +1,221 @@
+
+		if (activeFileName != null) {
+			return new RolloverDescriptionImpl(activeFileName, append, null,
+					null);
```
dead code below

brucehoff commented on services/repository/src/test/java/org/sagebionetworks/sweeper/log4j/TimeBasedRollingToS3PolicyTest.java:
```diff
@@ -0,0 +1,140 @@
+
+	@Test
+	public void testActivateOptions() {
+		activateOptions();
```
'activateOptions' is a misnomer.

RadicalZephyr committed:
Changes to logging configuration and code review fixes

Standardize the output format, use the new RollingPolicy and set
the logs to roll every minute.

RadicalZephyr commented:
@brucehoff Code review corrections have been made!

eric-wu commented on services/repository/src/main/java/org/sagebionetworks/sweeper/log4j/AmazonS3ProviderImpl.java:
```diff
@@ -0,0 +1,14 @@
+
+	@Override
+	public AmazonS3 getS3Client(String awsAccessKeyId, String awsAccessSecretKey) {
+		return new AmazonS3Client(new BasicAWSCredentials(awsAccessKeyId, awsAccessSecretKey));
```
Better reuse the same S3 client.

eric-wu commented on services/repository/src/main/java/org/sagebionetworks/sweeper/log4j/SweepAction.java:
```diff
@@ -0,0 +1,92 @@
+		return s3BucketName;
+	}
+
+	public void setS3BucketName(String s3BucketName) {
```
Consider removing the setters if they are not used so that the object is immutable.

RadicalZephyr committed:
Remove AmazonS3Provider, and moved to reusing a single S3Client
